enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotAllocation.scala

```diff
@@ -122,11 +126,16 @@ object SlotAllocation {
         case (Some(_), Some(right)) if comingFrom eq right =>
           val rhsSlots = resultStack.pop()
           val lhsSlots = resultStack.pop()
-          val result = allocate(current, nullable, lhsSlots, rhsSlots)
-          allocations += (current.assignedId -> result)
+          // NOTE: This assumes expressions on this logical plan are evaluated in the scope of the LHS
```

Comment:
Have you considered ValueHashJoin? It will evaluate expressions on both sides

Reply:
See: https://github.com/neo4j/neo4j/blob/3.4/enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlottedRewriter.scala#L95

Reply:
Good point!